lib/vapor/watch.ex

```diff
@@ -10,20 +10,34 @@ defmodule Vapor.Watch do
   end
 
   def init(state) do
+    state = Map.merge(%{error_count: 0, current_alarm: false}, state)
     schedule_refresh(state.plan)
     {:ok, state}
   end
 
   def handle_info(:refresh, %{plan: plan, store: store, layer: layer}=state) do
     case Provider.load(plan.provider) do
       {:ok, new_config} ->
+        # We don't mind if we repeatedly spam the logs when there's a problem,
+        # but if everything is okay, we ought not attempt to clear an alarm
+        # that might not exist because the attempt with spam the logs.
+        if state.current_alarm do
+          :alarm_handler.clear_alarm({:vapor, {layer, plan.provider}})
+        end
         :ok = Vapor.Store.update(store, layer, new_config)
         schedule_refresh(plan)
-        {:noreply, state}
+        {:noreply, %{state | current_alarm: false}}
 
-      {:error, _} ->
+      {:error, _reason} ->
```

Comment:
@keathley I think we're currently `raise`ing in a couple places, which certainly won't be caught by this. Given that we're dealing with potentially flaky sources, communication or read failures (after initial load) are not an exceptional event, IMO. Thoughts?

Reply:
I agree with that. If we're raising in other places we should probably remedy that I suppose. I'd have to look at the specific use cases I think to know what the right thing to do is.